### PR TITLE
Support peer filtering for booster-bitswap for endpoint

### LINKED ISSUE #847: booster-bitswap add ability to dynamically add Peer ID to denylist
For #709  adoption by Evergreen, need the ability to configure so that deny retrievals from certain peer id's. 

Need to be able to do this without restarting. 
Need to clearly document how to set and modify the configuration.

Reference -- https://www.notion.so/pl-strflt/Bitswap-Retrieval-Requirements-WIP-ed17a22b006e4ba79c9ca9ec85fcf93e

### PR DESCRIPTION
**Pull request by hannahhoward**
# Goals

fix #847 start on #954

# Implementation

- Extract periodic fetching logic from block filter to multifilter class
- Implement peer filter based on specs from @s0nik42
- Add peer filter endpoint as cli url

# For discussion

- Pending further discussion of additional retrieval configs with @s0nik42
- At somepoint booster bitswap will need a config file
- Otherwise GTG I think

### CODE REVIEW

File: cmd/booster-bitswap/filters/filters.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+func (mf *MultiFilter) Start(ctx context.Context) error {
+	mf.ctx, mf.cancel = context.WithCancel(ctx)
+	var cachedCopies []bool
+	for _, f := range mf.filters {
+		// open the cache file if it eixsts
+		cache, err := os.Open(f.cacheFile)
+		// if the file does not exist, synchronously fetch the list
+		if err != nil {
+			if !os.IsNotExist(err) {
+				return fmt.Errorf("fetching badbits list: %w", err)
```
Comment (dirkmc) [inline cmd/booster-bitswap/filters/filters.go:142]:
```suggestion
				return fmt.Errorf("fetching filter list: %w", err)
```

File: cmd/booster-bitswap/filters/filters.go [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+	for i, f := range mf.filters {
+		if cachedCopies[i] {
+			err := f.update()
+			if err != nil {
+				log.Error(err.Error())
+			}
+		}
+	}
+	updater := mf.clock.Ticker(UpdateInterval)
+	// call the callback if set
```
Comment (dirkmc) [inline cmd/booster-bitswap/filters/filters.go:192]:
```suggestion
        defer updater.Stop()
	// call the callback if set
```

**Review by dirkmc — APPROVED** [review]
Some small suggestions, but overall LGTM 👍

**Comment by dirkmc** [comment]
Please test with a local devnet before merging

**Comment by hannahhoward** [comment]
closing in favor of #1030

**Pull request closed without merging**